### DISCUSSION TITLE
Docusaurus: don't add routes automatically to the navbar

### DISCUSSION THREAD

**Discussion by kai687** [scalar/scalar#1896] [Ideas]
Hi, 

I really like the Docusaurus plugin, but it seems like each API reference gets added automatically to the navbar. 

I would like to use this for a project with about 11 APIs, which you can imagine won't work so great. I like to manually control the navbar, so in this case, I want an "API reference" group and add all the APIs to this group. 

Would it be possible to change the behavior of the plugin?
I imagine, something like if the label is empty, no entry will be added to the navbar.
(Currently, an entry with 'scalar' gets added).

**Comment by hanspagel** [comment]
Great idea! Let’s add an option or something: https://github.com/scalar/scalar/issues/2007

**Comment by htessaro** [comment]
That's a great idea. I'm thinking about using scalar in a Docusaurus project with 7 APIs, and my concern was exactly the same as @kai687 described.

It would be great to choose to add the links to the API pages as a dropdown menu, which we can use to customize the content. This behavior is available in the Docusaurus for documentation pages. Docusaurus documentation calls it [Navbar dropdown](https://docusaurus.io/docs/api/themes/configuration#navbar-dropdown).

**Comment by htessaro** [comment]
I have a complementary problem that I believe could be solved while developing this solution.

**Problem description:** When you add an API Reference, it automatically adds an item to the NavBar. However, as mentioned in the above comments, we don't have control over it. As a result, if we have a menu for mobile devices, we don't control the menu order. As presented in the example below, this results in the API Reference generated item appearing in the last position in the menu. 

![image](https://github.com/scalar/scalar/assets/85108199/46931372-bf1d-471c-88d0-19b68661efdc)

**Proposal**: It would be great to control the position of the API reference on the NavBar, in addition to whether it should appear there to avoid problems such as the one from the image.
